memory/src/main/scala/filodb.memory/format/vectors/Histogram.scala

```diff
@@ -480,6 +518,148 @@ final case class GeometricBuckets(firstBucket: Double,
   }
 }
 
+object Base2ExpHistogramBuckets {
+  // TODO: make maxBuckets default configurable; not straightforward to get handle to global config from here
+  val maxBuckets = 200
+}
+
+/**
+ * Open Telemetry Base2 Exponential Histogram Bucket Scheme.
+ * See https://github.com/open-telemetry/opentelemetry-proto/blob/7312bdf63218acf27fe96430b7231de37fd091f2/
+ *             opentelemetry/proto/metrics/v1/metrics.proto#L500
+ * for the definition.
+ * The bucket scheme is based on the formula:
+ * `bucket(index) = base ^ (index + 1)`
+ * where index is the bucket index, and `base = 2 ^ 2 ^ -scale`
+ *
+ * Negative buckets are not supported yet, so startIndexNegativeBuckets and numNegativeBuckets are always 0.
+ * They are here since the proto has it (but the client SDKs do not), and we want to be able to support it
+ * when the spec changes
+ *
+ * @param scale
+ * @param startIndexPositiveBuckets
+ * @param numPositiveBuckets
+ * @param startIndexNegativeBuckets negative buckets are not supported yet, so this is always 0
+ * @param numNegativeBuckets negative buckets are not supported yet, so this is always 0
+ */
+final case class Base2ExpHistogramBuckets(scale: Int,
+                                          startIndexPositiveBuckets: Int,
+                                          numPositiveBuckets: Int,
+                                          startIndexNegativeBuckets: Int = 0,
+                                          numNegativeBuckets: Int = 0
+                                        ) extends HistogramBuckets {
+  require(numPositiveBuckets <= Base2ExpHistogramBuckets.maxBuckets && numPositiveBuckets >= 0,
+    s"Invalid buckets: numPositiveBuckets=$numPositiveBuckets  maxBuckets=${Base2ExpHistogramBuckets.maxBuckets}")
+  require(numNegativeBuckets == 0 && startIndexNegativeBuckets == 0, "Negative buckets not supported yet")
+  require(startIndexPositiveBuckets > -500 && startIndexPositiveBuckets < 500)
```

Comment:
Very low and very high bucket indexes can lead to bucketTops that are out of range of Double, and can end up with zero. Hence the limits to avoid bad data from flowing in. I will add a comment. They will not conflict with 200 maxBucket default.